crates/shim-sgx/src/handler/key.rs

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use core::arch::asm;
+use core::mem::MaybeUninit;
+
+use sgx::enclu::EGETKEY;
+
+/// SGX derived key length in bytes
+pub const SGX_KEY_LEN: usize = 16;
+
+#[repr(C, align(16))]
+pub struct Response {
+    pub key: [u8; SGX_KEY_LEN],
+}
+
+#[repr(C, align(512))]
+pub struct Request {
+    pub name: Names,
+    pub policy: Policy,
+    pub isvsvn: u16,
+    pub cet_attr_mask: u8,
+    pub rsvd: u8,
+    pub cpusvn: [u8; 16],
+    pub attribute_mask: [u8; 16],
+    pub keyid: [u8; 32],
+    pub misc_mask: [u8; 4],
+    pub config_svn: u16,
+    pub rsvd2: [u8; 434],
+}
+
+impl Default for Request {
+    fn default() -> Self {
+        Self {
+            name: Names::EinittokenKey,
+            policy: Policy::MRSIGNER,
+            isvsvn: 0,
+            cet_attr_mask: 0,
+            rsvd: 0,
+            cpusvn: [0; 16],
+            attribute_mask: [0; 16],
+            keyid: [0; 32],
+            misc_mask: [0; 4],
+            config_svn: 0,
+            rsvd2: [0; 434],
+        }
+    }
+}
+
+#[allow(dead_code)]
```

Comment:
Because our code only uses `SealKey` the compiler moaned about the other stuff. Since I was implementing the Intel CPU Spec, I thought, that someone moving that stuff to a library might be happy.